### PR TITLE
fix error importing zip

### PR DESCRIPTION
**Pull request by thailekha**
Description
-extend testImport to test the content of the collection being imported
-replace for loop with forEach function since there were functions to be executed after the for loop constructed inside the for loop
-version bump

JIRA: https://issues.jboss.org/browse/RHMAP-13232

### CODE REVIEW

File: lib/importexport/helpers.js [hunk 1]
```diff
@@ -123,8 +123,7 @@ var _importZip = function(filePath, cb, retries){
     if (!zipEntries || zipEntries.length ===0){
       return cb('No zip entries found');
     }
-    for (var i=0; i<zipEntries.length; i++){
-      var zipEntry = zipEntries[i];
+    zipEntries.forEach(function(zipEntry) {
```
Comment (pb82) [inline lib/importexport/helpers.js]: What was the problem with the `for` loop here?

Comment (thailekha) [inline lib/importexport/helpers.js]: the problem was at https://github.com/thailekha/fh-db/blob/ecd09ede028bccf3dc05d3d69c114f6a5b937375/lib/importexport/helpers.js#L152 
a function is constructed here and it's dependent on `data` which is a pointer. If we use `for` loop then this pointer will always point the last item if zipEntries after the `for` loop finishes. `forEach` creates a new scope for every iteration I think.

File: lib/importexport/helpers.js [hunk 2]
```diff
@@ -123,7 +123,16 @@ var _importZip = function(filePath, cb, retries){
     if (!zipEntries || zipEntries.length ===0){
       return cb('No zip entries found');
     }
-    for (var i=0; i<zipEntries.length; i++){
+
+    for (var i = 0; i < zipEntries.length; i++){
+      if (zipEntries[i].isDirectory && zipEntries[i].entryName.indexOf('__MACOSX') > -1) {
+        zip.deleteFile(zipEntries[i])
```
Comment (fh-qube) [inline lib/importexport/helpers.js]: BLOCKER Add a semicolon at the end of this statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

File: lib/importexport/helpers.js [hunk 3]
```diff
@@ -149,10 +159,11 @@ var _importZip = function(filePath, cb, retries){
         return cb(msg);
       }
 
-      parsers[name] = function(parserCallback){
-        importHandler[fileType](data, parserCallback);
-      };
-
+      parsers[name] = function(parserData) {
```
Comment (fh-qube) [inline lib/importexport/helpers.js]: BLOCKER Add a semicolon at the end of this statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

File: lib/importexport/helpers.js [hunk 4]
```diff
@@ -123,7 +123,16 @@ var _importZip = function(filePath, cb, retries){
...
+
+    for (var i = 0; i < zipEntries.length; i++){
+      if (zipEntries[i].isDirectory && zipEntries[i].entryName.indexOf('__MACOSX') > -1) {
+        zip.deleteFile(zipEntries[i])
+      }
+    }
+
+    zipEntries = zip.getEntries();
+
+    for (var i = 0; i < zipEntries.length; i++){
```
Comment (fh-qube) [inline lib/importexport/helpers.js]: MAJOR Rename "i" as this name is already used in declaration at line 127. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2814)

**Comment by fh-qube** [comment]
SonarQube analysis reported 3 issues:
* BLOCKER 2 blocker
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request closed without merging**